src/checkfnis.cpp

```diff
@@ -129,8 +156,39 @@ QString CheckFNIS::generateIdentifier() const
   return QCryptographicHash::hash(flattenedList.join(",").toUtf8(), QCryptographicHash::Md5).toHex();
 }
 
+bool CheckFNIS::appIsFNIS(QString const &application, QString *fnisApp) const
+{
+  QStringList fnisBinaryList = m_MOInfo->findFiles("tools/GenerateFNIS_for_Users",
+    [] (const QString &fileName) -> bool { return fileName.endsWith("GenerateFNISforUsers.exe", Qt::CaseInsensitive); });
+
+  if (fnisBinaryList.count() == 0) {
+    // fnis seems not to be installed
+    qDebug("fnis not installed");
+    return false;
+  }
+
+  //As this is looking in the vfs, there can be precisely 0 or 1 instances of FNIS
+  //Stash the fnis name if requested.
+  if (fnisApp != nullptr) {
+    *fnisApp = fnisBinaryList.at(0);
+  }
+
+  return QString::compare(QDir::fromNativeSeparators(application),
+                          QDir::fromNativeSeparators(fnisBinaryList.at(0)),
+                          Qt::CaseInsensitive) == 0;
+}
+
 bool CheckFNIS::fnisCheck(const QString &application)
 {
+  if (!m_MOInfo->pluginSetting(name(), "enabled").toBool()) {
+    return true;
+  }
+
+  QString fnisApp;
+  if (appIsFNIS(application, &fnisApp)) {

```

Comment:
appIsFNIS returns false if fnis isn't installed so there is no early exit in this case and you will get a nasty error message further down.